medcat/utils/checkpoint.py

```diff
@@ -1,123 +1,166 @@
-        Args:
-            dir_path (str):
-                The path to the checkpoint directory.
-            steps (int):
-                The number of processed sentences/documents before a checkpoint is saved.
-                N.B.: A small number could result in error "no space left on device".
-            max_to_keep (int):
-                The maximum number of checkpoints to keep.
-                N.B.: A large number could result in error "no space left on device".
-            metadata (Optional[Dict]):
-                The extra training metadata need to be persisted.
-        """
+    def __init__(self, dir_path: str, *, steps: int = DEFAULT_STEP, max_to_keep: int = DEFAULT_MAX_TO_KEEP) -> None:
         self._dir_path = os.path.abspath(dir_path)
         self._steps = steps
         self._max_to_keep = max_to_keep
         self._file_paths: List[str] = []
         self._count = 0
-        self._metadata = metadata
         os.makedirs(self._dir_path, exist_ok=True)
 
     @property
     def steps(self) -> int:
         return self._steps
 
-    @steps.setter  # type: ignore
-    # [https://github.com/python/mypy/issues/1362]
-    @check_positive
+    @steps.setter
     def steps(self, value: int) -> None:
+        check_positive(lambda _: ...)(value)    # [https://github.com/python/mypy/issues/1362]
         self._steps = value
 
     @property
     def max_to_keep(self) -> int:
         return self._max_to_keep
 
-    @max_to_keep.setter  # type: ignore
-    # [https://github.com/python/mypy/issues/1362]
-    @check_positive
+    @max_to_keep.setter
     def max_to_keep(self, value: int) -> None:
+        check_positive(lambda _: ...)(value)    # [https://github.com/python/mypy/issues/1362]
         self._max_to_keep = value
 
     @property
     def count(self) -> int:
         return self._count
 
     @property
-    def metadata(self) -> Optional[Dict]:
-        return self._metadata
+    def dir_path(self) -> str:
+        return self._dir_path
+
+    @classmethod
+    def from_latest_training(cls: Type[T], base_dir_path: str) -> T:
+        r'''
+        Retrieve the latest checkpoint from the train directory.
+
+        Args:
+            base_dir_path (string):
+                The path to the directory containing checkpoint files
+        Returns:
+            A new checkpoint object
+        '''
+        if not os.path.isdir(base_dir_path):
+            raise ValueError(f"Checkpoint folder passed in does not exist: {base_dir_path}")
+        ckpt_dir_paths = os.listdir(base_dir_path)
+        if not ckpt_dir_paths:
+            raise ValueError("No existing training found")
+        ckpt_dir_paths.sort()
+        ckpt_dir_path = os.path.abspath(os.path.join(base_dir_path, ckpt_dir_paths[-1]))
+        checkpoint = cls.from_latest(dir_path=ckpt_dir_path)
+        return checkpoint
 
     @classmethod
-    def restore(cls, dir_path: str) -> "Checkpoint":
+    def from_latest(cls: Type[T], dir_path: str) -> T:
```

Comment:
Not sure what's going on here - but I rebased from originCAT-93 and I see:
    @classmethod
    def from_latest(cls: Type[T], dir_path: str) -> T:

 and 
    @classmethod
    def from_latest_training(cls: Type[T], base_dir_path: str) -> T:

that seem to do the same thing... 

Also we don't need a retrieve_latest, just retrieve that accepts a full checkpoint path